`eval` breaks `no-undef` and `no-unused-vars` rules

`babel-eslint@8.1.1`
`eslint@4.14.0`

#### `.eslintrc`
```yml
parser: babel-eslint
env:
  browser: true
rules:
  no-undef: error
  no-unused-vars: error
```

#### `index.js`
```js
const a = 1;
console.log(a);

eval('');
```

```sh
$ eslint index.js

/Users/jetbrains/IdeaProjects/eslint-repro/index.js
  1:7   error  'a' is not defined             no-undef
  1:7   error  'a' is defined but never used  no-unused-vars
  2:13  error  'a' is not defined             no-undef
```

After commenting the `eval` line, false positives disappear

--- a/lib/analyze-scope.js
+++ b/lib/analyze-scope.js
@@ -309,6 +309,7 @@ class Referencer extends OriginalReferencer {
 
 module.exports = function(ast, parserOptions) {
   const options = {
+    ignoreEval: true,
     optimistic: false,
     directive: false,
     nodejsScope:

--- a/test/non-regression.js
+++ b/test/non-regression.js
@@ -1874,4 +1874,16 @@ describe("verify", () => {
       []
     );
   });
+
+  it("ignore eval in scope analysis", () => {
+    verifyAndAssertMessages(
+      unpad(`
+        const a = 1;
+        console.log(a);
+        eval('');
+      `),
+      { "no-unused-vars": 1, "no-undef": 1 },
+      []
+    );
+  });
 });

